packages/ui/src/components/Checkbox/SCheckbox.vue

```diff
@@ -0,0 +1,270 @@
+    size: CheckboxSize.MEDIUM
+  },
+);
+
+const model = computed({
+  get(): CheckboxModelValue {
+    return props.modelValue
+  },
+  set(value: CheckboxModelValue) {
+    emit("update:modelValue", value);
+  }
+});
+
+const checked = computed((): boolean => {
+  if (typeof props.modelValue === 'boolean') {
+    return props.modelValue;
+  } else if (Array.isArray(props.modelValue)) {
+    return props.modelValue.includes(props.value);
+  } else {
+    return props.modelValue === props.trueValue;
+  }
+});
+</script>
+
+<template>
+  <label
+    :class="[
+      's-checkbox',
+      `s-checkbox--size-${size}`,
+      {
+        's-checkbox--bordered': border,
+        's-checkbox--checked': checked,
+        's-checkbox--disabled': disabled,
+        's-checkbox--indeterminate': indeterminate,
+      }
+    ]"
+  >
+    <div class="s-checkbox__content">
+      <span class="s-checkbox__input">
+        <input
+          :id="id"
+          :name="name"
+          :true-value="trueValue"
+          :false-value="falseValue"
+          :value="value"
+          :checked="checked"
+          :disabled="disabled"
+          :indeterminate="indeterminate"
+          v-model="model"
+          type="checkbox"
+          class="s-checkbox__original"
+        />
+        <span class="s-checkbox__inner" />
+      </span>
+      <span class="s-checkbox__label" v-if="$slots.default || label">
+        <slot>{{ label }}</slot>
+      </span>
+    </div>
+    <div class="s-checkbox__description" v-if="$slots.description">
+      <slot name="description" />
+    </div>
+  </label>
+</template>
+
+<style lang="scss">
+@mixin size($size, $padding, $labelOffset, $descriptionOffset) {
+  $base: calc(var(--s-size-#{$size}) / 2);
+
+  &.s-checkbox--size-#{$size} {
+    &.s-checkbox--bordered {
+      padding: $padding;
+    }
+
+    .s-checkbox__inner {
+      width: $base;
+      height: $base;
+    }
+
+    .s-checkbox__input + .s-checkbox__label {
+      margin-left: $labelOffset;
+    }
+
+    .s-checkbox__description {
+      margin-top: $descriptionOffset;
+    }
+  }
+}
+
+@mixin label-size($size, $fontSize, $lineHeight) {
+  &.s-checkbox--size-#{$size} {
+    .s-checkbox__label {
+      font-size: $fontSize;
+      line-height: $lineHeight;
+      letter-spacing: -0.02em;
+    }
+  }
+}
+
+@mixin checkmark-size($size, $top, $left, $width, $height) {
+  &.s-checkbox--size-#{$size} {
+    .s-checkbox__inner {
+      &::after {
+        top: $top;
+        left: $left;
+        width: $width;
+        height: $height;
+      }
+    }
+  }
+}
+
+.s-checkbox {
+  cursor: pointer;
+  display: inline-flex;
+  flex-flow: column nowrap;
+
+  @include size('mini', 7px 7px, 6px, 4px);
+  @include size('small', 7px 9px, 6px, 6px);
+  @include size('medium', 9px 9px, 8px, 8px);
+  @include label-size('mini', var(--s-font-size-2xs), 16px);
+  @include label-size('small', var(--s-font-size-xs), 18px);
+  @include label-size('medium', var(--s-font-size-sm), 20px);
```

Comment:
If you open Figma, you'll see such comment at font section:

```css
/* Paragraph / 3 - 12 */
```

That's a typography reference, and such styling includes **more that only a font size**.

When you need to apply some typography rule, please delegate it to Windi:

```scss
&--size-md {
  @apply s-ty-p3;
}
```

More about typography utils at `packages/theme/src/windi.ts`.